File: doc-examples/example-protobuf-java/src/test/java/example/LegacyReading.java
package example;

import io.micronaut.serde.annotation.Serdeable;
import io.micronaut.serde.protobuf.annotation.ProtoField;
import io.micronaut.serde.protobuf.annotation.ProtoType;

// tag::clazz[]
@Serdeable
public record LegacyReading(
    @ProtoField(1) String deviceId,
    @ProtoField(2) long recordedAt,
    @ProtoField(value = 3, type = ProtoType.SINT32) int temperatureMilliCelsius
) {
}
// end::clazz[]

File: doc-examples/example-protobuf-kotlin/src/test/kotlin/example/LegacyReading.kt
package example

import io.micronaut.serde.annotation.Serdeable
import io.micronaut.serde.protobuf.annotation.ProtoField
import io.micronaut.serde.protobuf.annotation.ProtoType

// tag::clazz[]
@Serdeable
data class LegacyReading(
    @field:ProtoField(1) val deviceId: String,
    @field:ProtoField(2) val recordedAt: Long,
    @field:ProtoField(value = 3, type = ProtoType.SINT32) val temperatureMilliCelsius: Int
)
// end::clazz[]
